### PR TITLE
Implement the Logical Assignment tests

### PR DESCRIPTION
**Pull request by afmenez**
Implementing the Logical Assignment tests.  
- I believe I tested all important cases.
- Added the results for Chrome and Firefox
- I am never sure about the significance field.  Is 'small' OK for this one?

### CODE REVIEW

File: data-esnext.js [hunk 1]
```diff
@@ -3881,6 +3881,65 @@ exports.tests = [
...
+        let c = 1;
+        a ||= 2;
+        b ||= 2;
+        c ||= 2;
+        return (a == 2) && (b == 2) && (c == 1);
+      */},
+      res: {
+        firefox76: false,
+        firefox77: firefox.nightly,
+        chrome80: false,
```
Comment (JLHwung) [inline data-esnext.js:4007]: Can confirm both `safari13` and `safaritp` does not support it either.

File: data-esnext.js [hunk 2]
```diff
@@ -3881,6 +3881,65 @@ exports.tests = [
...
+    {
+      name: '||=',
+      exec: function () {/*
+        let a;
+        let b = 0;
+        let c = 1;
+        a ||= 2;
+        b ||= 2;
+        c ||= 2;
+        return (a == 2) && (b == 2) && (c == 1);
```
Comment (JLHwung) [inline data-esnext.js]: nit: use `===` is preferred.

**Review by ljharb — APPROVED** [review]
"small" seems right to me. rebased and fixed the nits.

**Review by JLHwung — APPROVED** [review]

**Review by nicolo-ribaudo — COMMENTED** [review]
Depending on how much the tests should be specific, we could also test two differences these operators have from all the other assignment operators:
1. The short-circuiting behaviour (`true ||= ++i`) shouldn't increment `i`
2. Setters shouldn't be unnecessarily invoked:
    ```js
    var obj = { get x() { return true }, set x() { i++; } }
    obj.x ||= 2; // shouldn't increment i
    ```

**Review by nicolo-ribaudo — APPROVED** [review]

File: data-esnext.js (hunk 1 above)
Comment (afmenez) [reply data-esnext.js:4007]: added safari13 results, safaritp is inherited.

File: data-esnext.js (hunk 2 above)
Comment (afmenez) [reply data-esnext.js]: Done by @ljharb

**Review by chicoxyzzy — APPROVED** [review]

**Pull request merged**